lib/BackbeatConsumer.js

```diff
@@ -542,6 +547,68 @@ class BackbeatConsumer extends EventEmitter {
         return subscriptions.length > 0;
     }
 
+    /**
+     * Helper method to wait for consumers to be assigned to partitions before
+     * proceeding to send canary messages.
+     * @param {number} wait - current accumulated wait time
+     * @param {callback} cb - callback to invoke
+     * @return {undefined}
+     */
+    _waitForAssignment(wait, cb) {
+        setTimeout(() => {
+            const assignments = this._consumer.assignments();
+            if (assignments.length === 0) {
+                if (wait > 50000) {
+                    return cb(true);
```

Comment:
Can you log a warning here stating that it has timed out waiting to be assigned a partition?

Reply:
I added a debug log on L#585. I'll improve the message there